tools/licenses/lib/main.dart

```diff
@@ -2268,6 +2271,29 @@ Future<void> _collectLicensesForComponent(_RepositoryDirectory componentRoot, {
   sink.writeln('USED LICENSES:\n');
   final List<License> usedLicenses = licenses.where((License license) => license.isUsed).toList();
   final List<String> output = usedLicenses.map((License license) => license.toString()).toList();
+  for (int index = 0; index < output.length; index += 1) {
```

Comment:
I saw the "Sanity Checks" section at the bottom that this replaces, but what's the rationale for each of these? Like, why are we excluding each of these?

Also, should they be more generalized to catch more cases (e.g. if someone wraps/indents the input text differently, are these going to miss things?)

Reply:
They're cases that I ran into when writing the script a few years ago, which are indicative of an unexpected change in the licensing situation. I'll add a comment.